apps/judicial-system/backend/src/app/modules/case-table/caseTable.cellGenerators.ts

```diff
@@ -150,6 +156,128 @@ const generateRequestCaseStateTag = (c: Case, user: TUser): TagValue => {
+      return {
+        color: 'purple',
+        text: isDistrictCourtUser(user) ? 'Nýtt' : 'Sent',
+      }
+    case CaseState.RECEIVED:
+      return generateReceivedIndictmentStateTag(indictmentDecision, courtDate)
+    case CaseState.COMPLETED:
+      return generateCompletedIndictmentStateTag(
+        user,
+        indictmentRulingDecision,
+        indictmentReviewerId ? true : false,
+      )
+    case CaseState.WAITING_FOR_CANCELLATION:
+      return {
+        color: 'rose',
+        text: 'Afturkallað',
+      }
+    default:
+      return { color: 'white', text: 'Óþekkt' }
+  }
+}
+
+const generateReceivedIndictmentStateTag = (
+  indictmentDecision?: IndictmentDecision | null,
+  courtDate?: Date | string | null,
+): TagValue => {
+  switch (indictmentDecision) {
+    case IndictmentDecision.POSTPONING:
+    case IndictmentDecision.SCHEDULING:
+    case IndictmentDecision.COMPLETING:
+      return { color: 'mint', text: 'Á dagskrá' }
+    case IndictmentDecision.POSTPONING_UNTIL_VERDICT:
+      return { color: 'mint', text: 'Dómtekið' }
+    case IndictmentDecision.REDISTRIBUTING:
+      return { color: 'blue', text: 'Endurúthlutun' }
+    default:
+      return courtDate
+        ? { color: 'mint', text: 'Á dagskrá' }
+        : { color: 'blueberry', text: 'Móttekið' }
+  }
+}
+
+const generateCompletedIndictmentStateTag = (
+  user?: TUser,
+  indictmentRulingDecision?: CaseIndictmentRulingDecision | null,
+  isInReview?: boolean,
+): TagValue => {
+  if (isPublicProsecutionOfficeUser(user)) {
+    return {
+      color: isInReview ? 'mint' : 'purple',
+      text: isInReview ? 'Í yfirlestri' : 'Nýtt',
+    }
+  }
+  return generateIndictmentRulingDecisionTag(indictmentRulingDecision)
+}
+
+const generateIndictmentRulingDecisionTag = (
+  indictmentRulingDecision?: CaseIndictmentRulingDecision | null,
+): TagValue => {
+  switch (indictmentRulingDecision) {
+    case CaseIndictmentRulingDecision.FINE:
+      return { color: 'mint', text: 'Viðurlagaákvörðun' }
+    case CaseIndictmentRulingDecision.CANCELLATION:
+      return { color: 'rose', text: 'Niðurfelling' }
+    case CaseIndictmentRulingDecision.MERGE:
+      return { color: 'rose', text: 'Sameinað' }
+    case CaseIndictmentRulingDecision.DISMISSAL:
+      return { color: 'blue', text: 'Frávísun' }
+    case CaseIndictmentRulingDecision.RULING:
+      return { color: 'darkerBlue', text: 'Dómur' }
+    case CaseIndictmentRulingDecision.WITHDRAWAL:
+      return { color: 'rose', text: 'Afturkallað' }
+    default:
+      return { color: 'darkerBlue', text: 'Lokið' }
+  }
+}
+
+const generateIndictmentCaseType = (
+  indictmentSubtypes?: IndictmentSubtypeMap | null,
+): string | undefined => {
+  if (!indictmentSubtypes) return undefined
+
+  const labels =
+    getAllReadableIndictmentSubtypes(indictmentSubtypes).map(capitalize)
+
+  if (labels.length === 0) return undefined
+
+  return labels.length === 1 ? labels[0] : `${labels[0]} +${labels.length - 1}`
```

Comment:
Please note that the + should be on a separate line, so this should return a string list.